pkg/operation/botanist/secrets.go

```diff
@@ -600,6 +574,69 @@ func (b *Botanist) generateWantedSecrets(basicAuthAPIServer *secrets.BasicAuth,
 		},
 	})
 
+	if b.Shoot.KonnectivityTunnelEnabled {
+		// TODO: remove when all Gardener supported versions are >= 1.18
+		vpnSecretNamesToDelete := []string{"vpn-seed", "vpn-seed-tlsauth", "vpn-shoot"}
+		for _, secret := range vpnSecretNamesToDelete {
+			if err := b.K8sSeedClient.Client().Delete(context.TODO(), func(name, namespace string) *corev1.Secret {
+				return &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace},
+				}
+			}(secret, b.Shoot.SeedNamespace)); client.IgnoreNotFound(err) != nil {
+				return nil, err
+			}
+		}
```

Comment:
I would rather move out of this function as it should only generate the wanted secrets. Can you move it here: https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/secrets.go#L789-L791?